### PR TITLE
Remove hardcoded "htLeft" alignment from demos

### LINKED ISSUE #9419: Add support for RTL for main demo
### Description

The main demo at https://handsontable.com/docs/demo/ has hardcoded column headers alignment for left-to-right layout direction. It needs to be changed to add support for right-to-left. 

### Links

https://handsontable.com/docs/demo/

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR updates all demos (from the examples package) to add support for different settings for the layout direction option. From now on, all demos should look just as good as with left-to-right direction and right-to-left.

The PR shows that there is a need to expose a private API to the public. The methods `isRtl`, `isLtr`, and `getDirectionFactor` in the context of demos (and probably in the context of the user's apps) turned out to be very useful.

@kirszenbaum that three methods will land in docs so now is a chance to update their description.

### How has this been tested?

I tested the changes locally.

### Types of changes

- [x] New feature or improvement (non-breaking change which adds functionality)

### Related issue(s):
1. fixes #9419
2. https://github.com/handsontable/handsontable.com-v2/pull/150

### Affected project(s):
- [x] `handsontable`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Review by wszymanski — DISMISSED** [review]

**Comment by warpech** [comment]
This PR is required for the 12.0.0 release, but it does not need to be merged to the release branch, because the examples are deployed from the `develop` branch.

**Comment by budnix** [comment]
> This PR is required for the 12.0.0 release, but it does not need to be merged to the release branch, because the examples are deployed from the `develop` branch.

I think that PR is required to be merged to the release branch as it contains new methods that from now on are part of the public API. The docs need to be regenerated after merge as well.

File: .changelogs/9419.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "title": "Expose `isRtl`, `isLtr`, and `getDirectionFactor` methods to the public API.",
```
Comment (kirszenbaum) [inline .changelogs/9419.json:2]:
```suggestion
  "title": "Exposed the `isRtl()`, `isLtr()`, and `getDirectionFactor()` methods to the public API.",
```

File: handsontable/src/core.js [hunk 2]
```diff
@@ -136,9 +136,9 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
   this.rootElement.setAttribute('dir', rootElementDirection);
 
   /**
-   * Check if currently it is RTL direction.
+   * Checks if the table is rendered using right-to-left layout direction.
```
Comment (kirszenbaum) [inline handsontable/src/core.js:139]:
```suggestion
   * Checks if the grid is rendered using the right-to-left layout direction.
```

File: handsontable/src/core.js [hunk 3]
```diff
@@ -148,9 +148,9 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
   };
 
   /**
-   * Check if currently it is LTR direction.
+   * Checks if the table is rendered using left-to-right layout direction.
```
Comment (kirszenbaum) [inline handsontable/src/core.js:151]:
```suggestion
   * Checks if the grid is rendered using the left-to-right layout direction.
```

**Review by kirszenbaum — CHANGES_REQUESTED** [review]

**Review by wszymanski — APPROVED** [review]

**Review by kirszenbaum — APPROVED** [review]

**Pull request merged**